File: algorithms/Subarray Sum Equals K/Subarray Sum Equals K.js
	var subarraySum = function(nums, k) {
		const obj = {};
		let res = 0;
		let sum = 0;

		for (let i = 0; i < nums.length; i++) {

			sum += nums[i];
			if (sum == k) res++;

			if (obj[sum - k]) res += obj[sum - k];

			obj[sum] ? obj[sum] += 1 : obj[sum] = 1;
		}
		return res;
	};

File: algorithms/S/Subarray Sum Equals K/Subarray Sum Equals K.cpp

/********************************* Solution Using HashMap / Prefix ********************/

class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
       
        unordered_map<int, int> prefixSum {{0, 1}};
        
        int sum = 0;
        int numOfSubArr = 0;
        int size = nums.size();
        
        
        for (int index = 0; index < size; index++){
            
            sum += nums[index];
            if (prefixSum.find(sum-k) != prefixSum.end()) 
                numOfSubArr += prefixSum[sum-k];
            
            prefixSum[sum]++;
        }
        
        return numOfSubArr;
    }
};

